### PR TITLE
🐛added optional extention matching for dependency remap when bundling

### PR DESCRIPTION
**Pull request by dethstrobe**
Fix a bug when bundling not being able to find, very specifically `src/preact/compat.js` when remapping dependencies for react.

But this fix should also cover any remap key that is missing the js or ts extension.

### CODE REVIEW

File: build-system/tasks/helpers.js [hunk 1]
```diff
@@ -415,7 +415,7 @@ async function esbuildCompile(srcDir, srcFilename, destDir, options) {
    */
   function remapDependenciesPlugin() {
     const remaps = Object.entries(options.remapDependencies).map(
-      ([path, value]) => ({regex: new RegExp(`^${path}$`), value})
+      ([path, value]) => ({regex: new RegExp(`^${path}(\.[jt]s)?$`), value})
```
Comment (samouri) [inline build-system/tasks/helpers.js:418]:
```suggestion
      ([path, value]) => ({regex: new RegExp(`^${path}(\.[jt]sx?)?$`), value})
```

**Review by samouri — APPROVED** [review]
May want to add an `x` to the possible extensions. Or even simplify a bit with `(js|ts)x?)` etc.

File: build-system/tasks/helpers.js [hunk 2]
```diff
@@ -415,7 +415,7 @@ async function esbuildCompile(srcDir, srcFilename, destDir, options) {
    */
   function remapDependenciesPlugin() {
     const remaps = Object.entries(options.remapDependencies).map(
-      ([path, value]) => ({regex: new RegExp(`^${path}$`), value})
+      ([path, value]) => ({regex: new RegExp(`^${path}(\.[jt]sx?)?$`), value})
```
Comment (alanorozco) [inline build-system/tasks/helpers.js:418]: nit: unwrap regex parts for readability and discoverability

```suggestion
      ([path, value]) => ({regex: new RegExp(`^${path}(\.js|\.jsx|\.ts|\.tsx)?$`), value})
```

**Review by samouri — APPROVED** [review]

**Comment by samouri** [comment]
Can someone from @ampproject/wg-infra give this OWNERS approval?

File: build-system/tasks/helpers.js [hunk 3]
```diff
@@ -415,7 +415,10 @@ async function esbuildCompile(srcDir, srcFilename, destDir, options) {
    */
   function remapDependenciesPlugin() {
     const remaps = Object.entries(options.remapDependencies).map(
-      ([path, value]) => ({regex: new RegExp(`^${path}$`), value})
+      ([path, value]) => ({
+        regex: new RegExp(`^${path}(\.js|\.jsx|\.ts|\.tsx)?$`),
```
Comment (danielrozenberg) [inline build-system/tasks/helpers.js:419]: Supernit:
```suggestion
        regex: new RegExp(`^${path}(?:\.[jt]sx?)?$`),
```
(See https://regexper.com/#%5E%7Bpath%7D%28%3F%3A%5C.%5Bjt%5Dsx%3F%29%3F%24)

**Review by danielrozenberg — APPROVED** [review]

File: build-system/tasks/helpers.js (hunk 3 above)
Comment (dethstrobe) [reply build-system/tasks/helpers.js:419]: I love that regex.

But I think Alan has a point that if we make it ugly it'll be more discoverable when the next dev has to add one more extension or some other random thing.

Comment (danielrozenberg) [reply build-system/tasks/helpers.js:419]: Oh lol, didn't notice Alan's comment before :D
Fair enough, it's far from being a dealbreaker one way or the other

**Pull request merged**